vscode/src/gh-copilot/instructions.ts

```diff
@@ -4,295 +4,361 @@
 import { log } from "qsharp-lang";
 import * as vscode from "vscode";
 import { EventType, sendTelemetryEvent, UserFlowStatus } from "../telemetry";
-import { getActiveQdkDocumentUri } from "../programConfig";
 
-const codingInstructionsTitle = "# Q# coding instructions (updated April 2025)";
-
-const codingInstructionsFooter = `<!-- End: Q# coding instructions -->\n\n`;
-
-const codingInstructionsContent = `${codingInstructionsTitle}
-
-Follow these instructions when generating Q# code in .qs files,
-and Q# project folders that include a \`qsharp.json\` file.
-
-## Syntax
-
-When writing Q#, the \`for (i in 1..100)\` syntax is invalid, use \`for i in 1..100\` or
-\`for element in array[2...]\`.
-
-While many Q# operators are C-like, it uses \`or\` instead of \`||\` and \`and\` instead of \`&&\`.
-
-To extract values from a tuple, use destructuring via the \`let (item0, item1) = tupleValue;\` syntax.
-
-## Project structure
-
-### Single-file projects
-
-Q# files don't always need to exist in a project. A single \`.qs\` file can be compiled and
-run without a \`qsharp.json\` file. Prefer a single \`.qs\` file for simple programs.
-
-### Multi-file projects
+/**
+ * Command to update or create the Copilot instructions file for Q#.
+ * Shows a prompt to the user and updates the file if confirmed.
+ */
+export async function updateCopilotInstructions(
+  trigger: "Command" | "Project" | "Activation" | "ChatToolCall",
+  context: vscode.ExtensionContext,
+): Promise<string | undefined> {
+  const globalStateUri = context.globalStorageUri;
+  const userInvoked = trigger === "Command" || trigger === "Project";
+
+  if (isExtensionInstructionsConfigured(globalStateUri)) {
```

Comment:
If I'm reading correctly (and some of this code is a little hard to follow), then this is just checking that the instructions directory is listed in the user settings, and it doesn't check that the instructions are actually in that directory - is that correct? Is the assumption that if the directory is configured, then the instructions are always going to already be in there? Is that a safe assumption?